src/realm/object_converter.cpp

```diff
@@ -253,29 +244,210 @@ void InterRealmValueConverter::copy_dictionary(const Obj& src_obj, Obj& dst_obj,
+            handle_collection_in_mixed_mismatch(src_mixed, dst_mixed, type_List);
+            dst_obj.set_collection(m_dst_col, CollectionType::List);
+            Lst<Mixed> src_list{src_obj, m_src_col};
+            Lst<Mixed> dst_list{dst_obj, m_dst_col};
+            copy_list_in_mixed(src_list, dst_list, update_out);
+        }
+        else if (src_mixed.is_type(type_Set)) {
+            handle_collection_in_mixed_mismatch(src_mixed, dst_mixed, type_Set);
+            dst_obj.set_collection(m_dst_col, CollectionType::Set);
+            realm::Set<Mixed> src_set{src_obj, m_src_col};
+            realm::Set<Mixed> dst_set{dst_obj, m_dst_col};
+            // sets cannot be nested, so we just need to copy the values
+            copy_set(src_set, dst_set, update_out);
+        }
+        else if (src_mixed.is_type(type_Dictionary)) {
+            handle_collection_in_mixed_mismatch(src_mixed, dst_mixed, type_Dictionary);
+            dst_obj.set_collection(m_dst_col, CollectionType::Dictionary);
+            Dictionary src_dict{src_obj, m_src_col};
+            Dictionary dst_dict{dst_obj, m_dst_col};
+            copy_dictionary_in_mixed(src_dict, dst_dict, update_out);
+        }
+        else {
+            InterRealmValueConverter::ConversionResult converted_src;
+            auto dst_mixed = dst_obj.get_any(m_dst_col);
+            if (cmp_src_to_dst(src_mixed, dst_mixed, &converted_src, update_out)) {
+                if (converted_src.requires_new_embedded_object) {
+                    Obj new_embedded = dst_obj.create_and_set_linked_object(m_dst_col);
+                    track_new_embedded(converted_src.src_embedded_to_check, new_embedded);
+                }
+                else {
+                    dst_obj.set_any(m_dst_col, converted_src.converted_value);
+                }
             }
-            else {
-                dst_obj.set_any(m_dst_col, converted_src.converted_value);
+        }
+    }
+}
+
+bool InterRealmValueConverter::copy_list_in_mixed(const Lst<Mixed>& src_list, Lst<Mixed>& dst_list, bool* update_out)
+{
+    // utility functions
+    auto handle_set_copy = [&, this](size_t i) {
+        dst_list.insert_collection(i, CollectionType::Set);
+        auto n_src_set = src_list.get_set(i);
+        auto n_dst_set = dst_list.get_set(i);
+        copy_set(*n_src_set, *n_dst_set, update_out);
+    };
+
+    auto handle_list_copy = [&, this](size_t i) {
+        dst_list.insert_collection(i, CollectionType::List);
+        auto n_src_list = src_list.get_list(i);
+        auto n_dst_list = dst_list.get_list(i);
+        copy_list_in_mixed(*n_src_list, *n_dst_list, update_out);
+    };
+
+    auto handle_dict_copy = [&, this](size_t i) {
+        dst_list.insert_collection(i, CollectionType::Dictionary);
+        auto n_src_dict = src_list.get_dictionary(i);
+        auto n_dst_dict = dst_list.get_dictionary(i);
+        copy_dictionary_in_mixed(*n_src_dict, *n_dst_dict, update_out);
+    };
+
+    // main algorithm
```

Comment:
this isn't removing any excess elements that are in dest: `copy_list_in_mixed( {a, b}, {c, d, e} )` should produce `{a, b}` but currently it will produce `{a, b, c}`.

Reply:
This should now be fixed.